types/draft-js/index.d.ts

```diff
@@ -125,20 +126,20 @@ declare namespace Draft {
 
                 // Useful for managing special behavior for pressing the `Return` key. E.g.
                 // removing the style from an empty list item.
-                handleReturn?(e: SyntheticKeyboardEvent): DraftHandleValue,
+                handleReturn?(e: SyntheticKeyboardEvent, editorState: EditorState): DraftHandleValue,
 
                 // Map a key command string provided by your key binding function to a
                 // specified behavior.
-                handleKeyCommand?(command: EditorCommand): DraftHandleValue,
+                handleKeyCommand?(command: EditorCommand, editorState: EditorState): DraftHandleValue,
 
                 // Handle intended text insertion before the insertion occurs. This may be
                 // useful in cases where the user has entered characters that you would like
                 // to trigger some special behavior. E.g. immediately converting `:)` to an
                 // emoji Unicode character, or replacing ASCII quote characters with smart
                 // quotes.
-                handleBeforeInput?(chars: string): DraftHandleValue,
+                handleBeforeInput?(chars: string, editorState: EditorState): DraftHandleValue,
 
-                handlePastedText?(text: string, html?: string): DraftHandleValue,
+                handlePastedText?(text: string, html: string|undefined, editorState: EditorState): DraftHandleValue,
```

Comment:
i think that both html and editorState are optionals here so you don't need the `|undefined` and can go back to the `?` syntax?

Reply:
both `html` and `editorState` are always passed in DraftJs, no idea why the documentation shows html as optional... I think maybe because it can be `undefined`? https://github.com/facebook/draft-js/blob/eeb71f99c2d687b2094c7f43cc5989e8942e28dd/src/component/handlers/edit/editOnPaste.js#L105

Reply:
ah, I meant more along the lines of the function that you pass in here can optionally accept those params but it doesn't need to.. e.g. you can call `<RichEditor handlePastedText={(text) => 'handled'}/>`

Reply:
yeah, but in that case even `text` is optional 😄 
If I change this to `editorState?: EditorState` from a user perspective it looks like that parameter can be `undefined`, which is incorrect.

Reply:
ah yeah i was forgetting about this https://github.com/Microsoft/TypeScript/wiki/FAQ#why-are-functions-with-fewer-parameters-assignable-to-functions-that-take-more-parameters

Reply:
The `html?: string` syntax resolves the type of `html` to `string | undefined` but additionally allows you to omit the argument. If you get a valid result even if `text` is omitted/`undefined`, then all the parameters should probably be optional.

Reply:
>additionally allows you to omit the argument
☝️ that's not true, you can omit the argument with or without the `?`, check the link @willisplummer posted